### ISSUE TITLE
Start searching and rewards tasks on browser start after a set custom time.

### ISSUE THREAD

**Issue by TaherPatan** [mikeyaworski/ABS#35]
Hello, If you could add an option to start the searching and also after finishing the searches start the tasks as soon as the browser is opened it would be helpful, So that as soon as my browser is opened after a set time the searches and rewards should start. For eg: If I set the time to be 2 minutes so that whenever the browser is started, after 2 minutes it should check whether there are any searches or rewards and autocomplete them or if that is not possible then there should be something such that whenever the browser is started the searches and reward tasks are started after a set time, please add such functionality if possible.

**Comment by mikeyaworski** [comment]
So currently, there are scheduled searches:

![image](https://user-images.githubusercontent.com/10966214/99493521-58883f00-293d-11eb-83c1-c16a53af0a04.png)
![image](https://user-images.githubusercontent.com/10966214/99493539-650c9780-293d-11eb-8a8c-cfa0c9dca10a.png)

What this means is that if have searches scheduled for 5am, then if you open your browser at 6am, it will start searches as soon as it's opened. The scheduled time means searches will be started once per day, as long as you open your browser that day.

Are you suggesting that there should also be a delay before it opens the searches, or does this functionality work for you?

**Comment by TaherPatan** [comment]
> So currently, there are scheduled searches:
> 
> ![image](https://user-images.githubusercontent.com/10966214/99493521-58883f00-293d-11eb-83c1-c16a53af0a04.png)
> ![image](https://user-images.githubusercontent.com/10966214/99493539-650c9780-293d-11eb-8a8c-cfa0c9dca10a.png)
> 
> What this means is that if have searches scheduled for 5am, then if you open your browser at 6am, it will start searches as soon as it's opened. The scheduled time means searches will be started once per day, as long as you open your browser that day.
> 
> Are you suggesting that there should also be a delay before it starts the searches, or does this functionality work for you?

There should be a delay for the searches to start after I open the browser, sometimes I need to login before the searches should start, but in the above functionality it starts searching as soon as the browser is opened, could you add a delay function?

**Comment by mikeyaworski** [comment]
Yeah, when I get time, I can add an optional delay before the searches should start.

**Comment by TaherPatan** [comment]
> Yeah, when I get time, I can add an optional delay before the searches should start.

Is there anything currently that I can change in the code manually to add the delay?

**Comment by mikeyaworski** [comment]
You'll have to change this code:

https://github.com/mikeyaworski/ABS/blob/15b39924f93810622aa2c5a7d33ed941ec9b6db0/src/background-scripts/schedule.js#L37-L46

You'll need to add something along the lines of:

```
  if (lastSearch > getMidnightDate()) {
    when += constants.ONE_DAY_MILLIS;
  }
  // these are the new lines
  const TWO_MINS_IN_MILLIS = 2 * 60 * 1000;
  when = Math.max(when, Date.now() + TWO_MINS_IN_MILLIS);
```

Alternatively, you can just schedule the searches for much later in the day, when you know you'll have logged in already.
